### PR TITLE
Run tests in regtest mode via Travis CI

### PR DESCRIPTION
**Pull request by dexX7**
The developer preview of Omni Core is downloaded from Omni Chest, and the tests in regtest mode are executed via the test script.

### CODE REVIEW

**Comment by zathras-crypto** [comment]
Hey bud, at the meeting today it was requested we go for a release next week so we can get moving on Class C activation. I assume that means the Release Candidate @CraigSellars?

MetaDEx testing could then occur in parallel, and we could fix any identified MetaDEx bugs for a 0.0.10.1 release before activating that feature.

What do you think mate?

**Comment by dexX7** [comment]
Sounds fine to me.

Currently only the activation, and change log + release notes are for the release:

https://github.com/OmniLayer/omnicore/milestones/0.0.10

**Comment by dexX7** [comment]
https://github.com/OmniLayer/omnicore/issues/237 might be picked up though.

**Comment by dexX7** [comment]
So right now I don't recommend to merge this, because the tests will never pass (due to the unfixed issues of the binary).

We could upload and use a binary build from the current `omnicore-0.0.10` tip, which includes the fixes, so that the tests pass, as temporariy solution, until the next "official" release is out. What do you think @msgilligan?

Also: do you have the rank to enable Travis CI for this repo? Otherwise I'll send @achamely an email. :)

File: test-omni-integ-regtest.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ trap cleanup EXIT
 BTCD=copied-artifacts/src/omnicored
 DATADIR=$HOME/.bitcoin
 LOGDIR=logs
-MSCLOG=/tmp/mastercore.log
+MSCLOG=/tmp/omnicore.log
```
Comment (dexX7) [inline test-omni-integ-regtest.sh]: @msgilligan: does this interfer with the Jenkins setup (i.e. does Jenkins expect `mastercore.log`)?

Comment (dexX7) [inline test-omni-integ-regtest.sh]: I left it how it was for now.

**Comment by msgilligan** [comment]
> We could upload and use a binary build from the current omnicore-0.0.10 tip, which includes the fixes, so that the tests pass, as temporariy solution, until the next "official" release is out. What do you think @msgilligan?

That sounds fine to me. It would also be nice if we could run a (subset of) tests against a previous version 0.0.9 if we can add conditionals for any features that require the newer version.

> Also: do you have the rank to enable Travis CI for this repo?

I don't know, but Adam took care of it. (Thanks @achamely !)  I need to focus on other things today, but will try to setup Travis for bitcionj-addons over the weekend.

File: test-omni-integ-regtest.sh (hunk 1 above)
Comment (msgilligan) [inline test-omni-integ-regtest.sh]: @dexX7 Is there still a log created in `/tmp`? Can we just remove that entirely. (It's used to create a symlink from $LOGDIR to a log file in /tmp.)

Comment (msgilligan) [inline test-omni-integ-regtest.sh]: I think I added the symlink so the log could be viewed in the workspace [here](https://ci.omni.foundation/job/omni-integ-regtest/ws/logs/) -- I can't even remember. We can probably remove that variable and the symlink.

**Comment by dexX7** [comment]
I'll tackle it soon~ish. The binaries I build locally were incompatible, see:

https://travis-ci.org/dexX7/OmniJ/builds/83303824#L136-L138

That's appearingly caused because I used gcc 4.9.0. The cleanest way is probably the Gitian build route in a VM, which uses Ubuntu 12.04 as build VM. Once I find some time, I'll set it up.

> That sounds fine to me. It would also be nice if we could run a (subset of) tests against a previous version 0.0.9 if we can add conditionals for any features that require the newer version.

We could do this, but what do you suggest in particular? Since we don't have a proper release, it would basically be hand-picking a version + tests. The last "preview" (currently hosted on Omni Chest) was https://github.com/OmniLayer/omnicore/commit/d62b0a964924b3804fb5c6805b37e401dd20052c, and here is an overview for the changes (see at the bottom, "Since d62b0a9, the third developer preview"):

https://github.com/OmniLayer/omnicore/issues/96#issuecomment-119294812

**Comment by dexX7** [comment]
Ready.

The build is based on https://github.com/OmniLayer/omnicore/commit/43a7014c5700d041c0630b2494ff3e194db55a8e (most recent).

**Pull request merged**